test/integration_test/backup_test.go

```diff
@@ -0,0 +1,137 @@
+// +build integrationtest
+
+package integrationtest
+
+import (
+	"fmt"
+	"testing"
+
+	storkv1 "github.com/libopenstorage/stork/pkg/apis/stork/v1alpha1"
+	"github.com/portworx/sched-ops/k8s"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	instanceID = "basic"
+	appKey     = "mysql-app-to-backup"
+)
+
+func testBackup(t *testing.T) {
+	t.Run("backupTest", backupRestoreDeleteTest)
+}
+
+var backupLocationMap = map[string]storkv1.BackupLocationType{
+	"s3":     storkv1.BackupLocationS3,
+	"azure":  storkv1.BackupLocationAzure,
+	"google": storkv1.BackupLocationGoogle,
+}
+
+func triggerBackupRestoreTest(
+	t *testing.T,
+	instanceID string,
+	appKey string,
+	appBackupKey []string,
+	appRestoreKey []string,
+) {
+	var err error
+	// Reset config in case of error
+	defer func() {
+		err = setRemoteConfig("")
+		require.NoError(t, err, "Error resetting remote config")
+	}()
+
+	ctxs, _ := createAppForBackup(t, instanceID, appKey)
+	logrus.Infof("App created. Starting backup.")
+
+	// Create backuplocation here programatically using config-map that contains name of secrets to be used, passed from the CLI
+	err = createBackupLocation(t, "greatdane", appKey+"-"+instanceID, "s3", "secret-config")
+	require.NoError(t, err, "Error creating backuplocation")
+
+	err = schedulerDriver.AddTasks(ctxs[0],
+		scheduler.ScheduleOptions{AppKeys: appBackupKey})
+	require.NoError(t, err, "Error backing-up apps")
+	err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "Error waiting for back-up to complete.")
+
+	logrus.Infof("Backup completed. Starting Restore.")
+
+	err = schedulerDriver.AddTasks(ctxs[0],
+		scheduler.ScheduleOptions{AppKeys: appRestoreKey})
+
+	require.NoError(t, err, "Error restoring apps")
+	err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "Error waiting for restore to complete.")
+
+	logrus.Infof("Restore completed.")
+
+	destroyAndWait(t, ctxs)
+}
+
+func createBackupLocation(
+	t *testing.T,
+	name string,
+	namespace string,
+	locationtype string,
+	configMapName string,
+) error {
+	configMap, err := k8s.Instance().GetConfigMap(configMapName, "default")
+	require.NoError(t, err, fmt.Sprintf("Failed to get config map  %s", configMapName))
+
+	secretName := configMap.Data["secret_name"]
+
+	secretObj, err := k8s.Instance().GetSecret(secretName, "default")
+	require.NoError(t, err, fmt.Sprintf("Failed to get secret %s", secretName))
```

Comment:
Don't need to use `fmt.Sprintf()`, `NoError` takes message and params. Same for other places

Reply:
Fixed.